tap_google_analytics/ga_client.py

```diff
@@ -181,17 +184,29 @@ def lookup_data_type(self, type, attribute):
     def process_stream(self, stream):
         try:
             records = []
-            report_definition = self.generate_report_definition(stream)
-            nextPageToken = None
+            bookmark = singer.get_bookmark(self.state, stream.get("name"), "report_start_date")
+            if bookmark:
+                start_date = datetime.strptime(bookmark, "%Y-%m-%d") + timedelta(days=1)
```

Comment:
Just asking because date boundaries are always tricky. Is it possible to ever miss data because we advance start date by a day? for example, if the previous sync happened in the middle of February 1st (12pm) the the bookmark will be for Feb1st, but technically there is still 12 hours left of data to be collected. 

The safe option here would be to never add `timedelta(days=1)` and this way we'd replicate the last day's data twice, which is ok from Airbyte's perspective.

Reply:
Singer tap syncing process doesn't include [end date](https://github.com/airbytehq/pipelinewise-tap-google-analytics/blob/e6d1d3d0b0d27f037f6f2e767e700926f55beae6/tap_google_analytics/__init__.py#L149). So when sync happens in the middle of February 1st, records will be returned only for January 31st.

Reply:
sounds good